src/analysis/rMTA/rMTA.m

```diff
@@ -225,28 +225,17 @@
 fprintf('Step 2 in progress: MOMA\n');
 timerVal = tic;
 
-% Create the CPLEX model
-% variables
-v = 1:length(model.rxns);
-ctype(v) = 'C';
-n_var = v(end);
-
-% Objective fuction
-% linear part
-c(v) = -2*Vref;
-% quadratic part
-Q = 2*eye(n_var);
-
-cplex_model_moma.A = model.S;
-cplex_model_moma.lb = model.lb;
-cplex_model_moma.ub = model.ub;
-cplex_model_moma.lhs = zeros(size(model.mets));
-cplex_model_moma.rhs = zeros(size(model.mets));
-cplex_model_moma.obj = c;
-cplex_model_moma.Q = Q;
-cplex_model_moma.sense = 'minimize';
-cplex_model_moma.ctype = ctype;
-fprintf('\tcplex model for MOMA built\n');
+QPproblem = struct();
```

Comment:
What about using `buildLPProblemFromModel`?
That way you will Cover C and E matricies without worring about them. 
In this instance it would look like:
```
[nRxns,nMets] = size(model.S);
if isfield(model,E)
 nVars = size(model.E,2);
else 
 nVars = 0;
end
if isfield(model,E)
 nCtrs = size(model.C,1);
else 
 nCtrs = 0;
end
QPproblem = buildLPProblemFromModel(model);
QPproblem.c(1:nRxns) = -2*Vref;
QPproblem.F = [2*speye(nRxns),sparse(nRxns,nVars),sparse(nVars,nRxns+nVars)
QPproblem.osense = 1;
```
Also, is it intentional, that you reset any potential settings made in model.b /model.csense ?

Reply:
Hi! 
It seems prety interesting, but I am just performing MOMA analysis to obtain the resulting flux distribution after deleting some reactions. 
Furthermore, I have not taken into account C and E matrix for the MIQP problems, so it would be not accurate to use them in the MOMA formulation.

On the other hand, I have looked at the help of `buildLPProblemFromModel` and it is not clear to me how to use this tool. I suppose that E matrix are additional variables that are after the flux variables, that C matrix does not include S matrix, and that the order of the constrains must be S, C, E and D. As I am not sure how to use it correctly I prefer not to use it.

Reply:
Hi,
There is a tutorial for these additional Constraints/Variables (https://github.com/opencobra/COBRA.tutorials/blob/master/reconstruction/constrainingModels/tutorial_AdditionalConstraintsAndVariables.m)
`buildLPProblemFromModel` essentially builds an LP from the given model with
```
A = [S , E;...
       C , D];
```
and the appropriate associated fields (bounds, constraint senses etc). Overall its just a helper function to make code less dependent on the internal model structure.

Reply:
All one can do is suggest such changes, they are not required. In this case, the code to build the problem structure does the same thing, so we should not delay this PR because of it.

Reply:
Thanks @tpfau !

I was not able to find the tutorial for the `buildLPProblemFromModel ` function. I will try to use it with the next upgrade to the code.

Reply:
The reason I'm suggesting it is to avoid getting new functions in which do not support the whole COBRA structure, i.e. do calculations which might miss properties set in the model.

Reply:
I have uploaded a commit using `buildLPproblemFromModel`  function for the moma part of the algorithm. I have also changed the name of one function, from `MTA_model` to  `buildMTAproblemFromModel`, to have some similarities in the nomenclature of the code